server/athenian/api/controllers/features/github/pull_request_metrics.py

```diff
@@ -654,3 +654,70 @@ def _analyze(self, facts: pd.DataFrame, min_time: datetime, max_time: datetime,
         sizes = facts["size"].values.astype(object)
         sizes[self._calcs[0].peek == np.array(None)] = None
         return sizes
+
+
+class StagePendingDependencyCalculator(WithoutQuantilesMixin, SumMetricCalculator[int]):
+    """Common dependency for stage-pending counters."""
+
+    dtype = int
+
+    def _analyze(self, facts: np.ndarray, min_time: datetime, max_time: datetime,
+                 **kwargs) -> np.ndarray:
+        merged_mask, approved_mask, frr_mask = (
+            (v := facts[c].values) == v
+            for c in ("merged", "approved", "first_review_request"))
+
+        stage_indexes = {}
+        other = ~facts["done"].values
+        stage_indexes["release"] = np.where(merged_mask & other)[0]
+        other &= ~merged_mask
+        stage_indexes["merge"] = np.where(approved_mask & other)[0]
+        other &= ~approved_mask
+        stage_indexes["review"] = np.where(frr_mask & other)[0]
+        other &= ~frr_mask
+        stage_indexes["wip"] = np.where(other)[0]
+        self._stage_indexes = stage_indexes
+
+        return np.array([], dtype=object)
```

Comment:
I am not sure if it will work, but we can try:

```python
return np.array([stage_indexes])
```

and then

```python
result[self._calcs[0].peek[0][self.stage_name]] = 1
```

Reply:
...we'll also have to set `dtype = object` in `StagePendingDependencyCalculator`

Reply:
this is much cleaner 👍